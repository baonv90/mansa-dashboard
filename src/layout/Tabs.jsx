import React, { useState, useCallback } from 'react';
import styled from 'styled-components';


const TabsContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin-top: 1rem;
`;

const Header = styled.div`
  display: flex;
  align-items: center;
  min-height: 100px;
  overflow-y: scroll;
`;

// Tabs component display a TabNav component and its content based on the selected Tab
const Tabs = ({ children, TabNav }) => {
  const [selectedTab, setSelectedTab] = useState(0);
  const renderTitles = useCallback((items) => 
    items.map(({ props }, index) => (
      <TabNav
        key={index}
        index={index}
        isSelected={selectedTab === index}
        setSelectedTab={setSelectedTab}
        {...props}
      />
    )
  ), [setSelectedTab, selectedTab]);

  return (
    <TabsContainer>
      <Header>
        {renderTitles(children)}
      </Header>
      {children[selectedTab]}
    </TabsContainer>
  )
};

export default Tabs;
