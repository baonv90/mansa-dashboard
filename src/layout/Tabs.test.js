import renderer from 'react-test-renderer';
import Tabs from './Tabs';

describe('layout > Tabs', () => {
  const children = [
    <div key={1}>1</div>,
    <div key={2}>2</div>,
    <div key={3}>3</div>
  ];
  const TabNav = ({ title }) => <span>{title}</span>;

  it('should match snapshot', () =>  {
    const component = renderer.create(
      <Tabs TabNav={TabNav}>
        {children}
      </Tabs>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const els = component.root.findAllByType(TabNav);
    expect(els).toHaveLength(children.length);
  });
});
