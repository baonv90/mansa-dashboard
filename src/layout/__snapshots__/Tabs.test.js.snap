// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout > Tabs should match snapshot 1`] = `
<div
  className="sc-bdnxRM fvtCJR"
>
  <div
    className="sc-gtsrHT fSDNVI"
  >
    <span />
    <span />
    <span />
  </div>
  <div>
    1
  </div>
</div>
`;
